src/components/m-datepicker/_template.js

```diff
@@ -2,40 +2,97 @@ import html from 'nanohtml';
+
+  if (!higherEndYear) {
+    higherEndYear = date.getFullYear() + DEFAULT_NEW_YEARS;
+  }
+
+  if (startMonth !== TODAY) {
+    date.setMonth(startMonth);
+  }
+
+  if (startYear !== TODAY) {
+    date.setFullYear(startYear);
+  }
+
+  const year = date.getFullYear();
+
+  if (year < lowerEndYear) {
+    return {
+      year: lowerEndYear,
+      month: 0,
+    };
+  }
+
+  if (year > higherEndYear) {
+    return {
+      year: higherEndYear,
+      month: 11,
+    };
+  }
+
+  return {
+    year,
+    month: date.getMonth(),
+  };
+};
+
 export default ({
   classes,
   buttonOk,
   buttonCancel,
   locale = 'en-uk',
-  maxYears = 50,
-  futureYears = true,
-}) => html`
-  <article class="${classes} m-datepicker__article">
-      <div class="m-datepicker__dropdown">
-        <axa-dropdown class="m-datepicker__dropdown__month" size="sm" value="${new Date().getMonth()}" items="
-          ${JSON.stringify(getAllLocaleMonthsArray(locale).map((month, index) => {
-            return { name: month, url: '#', value: index };
-          }))}
-        ">
+  // value = 'iso',
+  startYear = TODAY,
+  startMonth = TODAY,
+  lowerEndYear,
+  higherEndYear,
+}) => {
+  const startDate = getStartDate({ lowerEndYear, higherEndYear }, startYear, startMonth);
+  const specificYears = getSpecificYears({ lowerEndYear, higherEndYear });
+  return html`
+    <article class="${classes} m-datepicker__article">
+      <div class="">
+        <axa-dropdown data-month="true" class="m-datepicker__dropdown m-datepicker__dropdown__month js-datepicker__dropdown__month"
+          size="sm" value="${startDate.month}"
+          items="${JSON.stringify(getAllLocaleMonthsArray(locale).map((month, index) => ({
+            name: month, url: '#', value: index,
+          })))}">
         </axa-dropdown>
-        <axa-dropdown class="m-datepicker__dropdown__year" size="sm" value="${new Date().getFullYear()}" items="
-          ${JSON.stringify(getSpecificYears(maxYears, futureYears).map((year) => {
-            return { name: year, url: '#', value: year };
-          }))}
-        ">
+        <axa-dropdown data-year="true" class="m-datepicker__dropdown m-datepicker__dropdown__year js-datepicker__dropdown__year"
+          size="sm" value="${startDate.year}"
+          items="${JSON.stringify(specificYears.map(year => ({
+            name: year, url: '#', value: year,
+          })))}">
         </axa-dropdown>
       </div>
       <div class="m-datepicker__weekdays">
-      ${getLocalWeekdayArray(locale).map(day => html`<p class="m-datepicker__weekdays__day">${day}</p>`)}
+        ${getLocalWeekdayArray(locale).map(day => html`<p class="m-datepicker__weekdays__day">${day}</p>`)}
       </div>
-      <div class="m-datepicker__body" data-calendar-area="month"></div>
+      <axa-datepicker-body allowed-years="${JSON.stringify(specificYears)}" year="${startDate.year}" month="${startDate.month}" class="js-datepicker__datepicker-body" locale="${locale}"></axa-datepicker-body>
```

Comment:
BTW. we still have this problem of inline `JSON.stringifiy` usage ... other topic.